src/Simplex/Chat.hs

```diff
@@ -2482,15 +2481,16 @@ processAgentMessage (Just user@User {userId}) corrId agentConnId agentMessage =
         forM_ r . uncurry $ probeMatch c1
 
     probeMatch :: Contact -> Contact -> Probe -> m ()
-    probeMatch c1@Contact {profile = p1} c2@Contact {profile = p2} probe =
-      when (fromLocalProfile p1 == fromLocalProfile p2) $ do
+    probeMatch c1@Contact {contactId = ctId1, profile = p1} c2@Contact {contactId = ctId2, profile = p2} probe =
+      when (profilesMatch (fromLocalProfile p1) (fromLocalProfile p2)) $ do
         void . sendDirectContactMessage c1 $ XInfoProbeOk probe
-        mergeContacts c1 c2
+        unless (ctId1 == ctId2) $ mergeContacts c1 c2
```

Comment:
unless can be in mergeContacts, then it doesn't need to be repeated/pattern matched in all places